bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/ui/navigator/CSymbolsContentProvider.java

```diff
@@ -107,10 +116,18 @@ protected void refreshTreeContentFromLS(IFile file) {
 				symbolsModel.update(response);
 			}).join();
 		} else {
+			temporaryLoadedDocument = false;
 			symbolsModel.setUri(file.getLocationURI());
 			symbolsModel.update(null);
 		}
-
+		if (temporaryLoadedDocument) {
+			try {
+				FileBuffers.getTextFileBufferManager().disconnect(file.getFullPath(), LocationKind.IFILE,
+						new NullProgressMonitor());
+			} catch (CoreException e) {
+				Platform.getLog(getClass()).log(e.getStatus());
```

Comment:
Doing e.getStatus() can lose where the exception was thrown from as it means the stack trace of e isn't preserved. So in the case where someone throws `new CoreException(Status.error("just a message"))` all that gets logged is "just a message" with no stack trace. If the called functions throws `new CoreException(Status.error("just a message", new Exception()))` then the correct stack trace will be included. At line 108 it looks a bit better to me because no information is lost.

Reply:
Replied there, @jonahgraham is right, it *was* better, but now we have counterexamples

Reply:
Not sure it is important, but when _was_ it better?

Reply:
It was better In all good times, when `CoreException` and `IStatus` were used properly to wrap "generic" exception for better error reporting and analysis. Please have a look at https://github.com/eclipse-platform/eclipse.platform/blob/master/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/IResourceStatus.java to have an illustration of how it was and what it was for.
Now, when people are struggling with Platform API signatures using modern static methods instead of following Platform design, your arguments look much more relevant.

Reply:
@ruspl-afed or @jonahgraham Are there any open issues here from your side?